pkg/util/rbac.go

```diff
@@ -169,6 +169,13 @@ func CreateOrUpdateClusterRole(client kubeclient.Interface, clusterRole *rbacv1.
 			return fmt.Errorf("unable to create RBAC clusterrole: %v", err)
 		}
 
+		existClusterRole, err := client.RbacV1().ClusterRoles().Get(context.TODO(), clusterRole.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		clusterRole.ResourceVersion = existClusterRole.ResourceVersion
```

Comment:
I'm ok to get the resource again before updating. I wonder the reason why you change this, do you find anything wrong?

Reply:
I find many creating/updating functions are not inconsistent(some with resourceVersion updating, some without).

The beneficial is it makes update operation atomization.

Reply:
`func CreateOrUpdateClusterRole()` and ``func CreateOrUpdateClusterRoleBinding()` only are used by `register` command, and they also are implemented at

https://github.com/karmada-io/karmada/blob/f7cac365d743e5e40493f9ad90352f30123f7f1d/pkg/karmadactl/util/idempotency.go#L148-L150

https://github.com/karmada-io/karmada/blob/f7cac365d743e5e40493f9ad90352f30123f7f1d/pkg/karmadactl/util/idempotency.go#L172-L174

Could you please help to cleanup them?

Reply:
So I should delete the redundant code in `karmada/pkg/karmadactl/util/idempotency.go`?

Reply:
> So I should delete the redundant code in karmada/pkg/karmadactl/util/idempotency.go?

No, cleanup  `pkg/util/rbac.go`, because they are only used by karmadactl.